File: Python/dir_swc_labels/lib/menu/select_subregion.py
def select_source_subregion():
    subregions = ['DG', \
                  'CA3', \
                  'CA2', \
                  'CA1', \
                  'SUB', \
                  'EC']
                  
    reply = ''
    
    # main loop to display menu choices and accept input
    # terminates when user chooses to exit
    while (not reply):
    
        try:
            print("\033c"); # clear screen
                
            ## display menu ##
            print 'Please select your subregion of interest from the selections below:\n'
                
            print '     1) Dentate Gyrus (DG)'
            
            print '     2) CA3'
            
            print '     3) CA2'
            
            print '     4) CA1'
            
            print '     5) Subiculum (SUB)'
            
            print '     6) Entorhinal Cortex (EC)'
            
            print '     a) All subregions'
            
            print '     !) Exit'

            reply = raw_input('\nYour selection: ')
                
            ## process input ##
            if reply == '!':
                return('!')
            elif reply == 'a':
                return('All')
            else:
                num = int(reply)
                if ((num > 0) & (num <= 6)):
                    return(subregions[num-1])
                else:
                    reply = ''
                
        except ValueError:
            print 'Oops! That was not a valid number. Please try again ...'

File: MATLAB/dir_swc_labels_v2/lib/menu/select_subregion.m
function subregion = select_subregion()

    subregions = { 'DG', ...
                   'CA3', ...
                   'CA2', ...
                   'CA1', ...
                   'SUB', ...
                   'EC' };

    reply = [];
    
    % main loop to display menu choices and accept input
    % terminates when user chooses to exit
    while (isempty(reply))
        %% display menu %%
        
        clc;
        
        strng = sprintf('Please select your subregion of interest from the selections below:\n');
        disp(strng);

        strng = sprintf('     1) Dentate Gyrus (DG)');
        disp(strng);
        
        strng = sprintf('     2) CA3');
        disp(strng);
        
        strng = sprintf('     3) CA2');
        disp(strng);
        
        strng = sprintf('     4) CA1');
        disp(strng);
        
        strng = sprintf('     5) Subiculum (SUB)');
        disp(strng);
        
        strng = sprintf('     6) Entorhinal Cortex (EC)');
        disp(strng);
        
        strng = sprintf('     a) All subregions');
        disp(strng);
        
        disp('     !) Exit');
        
        reply = input('\nYour selection: ', 's');

        
        %% process input %%
        
        if strcmp(reply, '!')
            subregion = '!';
        elseif strcmp(reply, 'a')
            subregion = 'All';
        else 
            num = str2double(reply);
            
            if ((num > 0) && (num <= 6))
                subregion = subregions{num};
            else
                reply = [];
            end % if num
        end
        
    end % while loop

end % select_file()